scripts/upload_dos_opportunities_email_list.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+
+"""
+Description:
+    For every lot the script will:
+    - find a list of emails for all suppliers and their users from the API
+    - add new emails to mailchimp list if they're not there. If the user previously unsubscribed themselves
+    from the list, they will not be subscribed when the script runs.
+
+    Please be aware that sometimes there can be a delay (up to a few minutes) in adding new addresses
+    to a mailchimp user list after an API call has been made. Unfortunately we couldn't find mailchimp documentation
+    regarding this.
+
+    If in the command line you provide an environment other than production the script will run on test lists
+    that we have setup on mailchimp.
+
+Usage:
+    upload_dos_opportunities_email_list.py <stage> <api_token> <mailchimp_username> <mailchimp_api_key>
+
+Example:
+    upload_dos_opportunities_email_list.py preview myToken user@gds.gov.uk 7483crh87h34c3
+
+"""
+
+import sys
+
+from docopt import docopt
+from dmapiclient import DataAPIClient
+from dmutils.email.dm_mailchimp import DMMailChimpClient
+
+sys.path.insert(0, '.')
+from dmscripts.helpers.env_helpers import get_api_endpoint_from_stage
+from dmscripts.upload_dos_opportunities_email_list import main
+from dmscripts.helpers import logging_helpers
+from dmscripts.helpers.logging_helpers import logging
+
+
+logger = logging_helpers.configure_logger({'dmapiclient': logging.INFO})
+
+
+if __name__ == '__main__':
+    arguments = docopt(__doc__)
+
+    lots = [
+        {
+            "lot_slug": "digital-specialists",
+            "list_id": "prod_list_id" if arguments['<stage>'] == "production" else "07c21f0451",
+            "framework_slug": "digital-outcomes-and-specialists-2"
+        },
+        {
+            "lot_slug": "digital-outcomes",
+            "list_id": "prod_list_id2" if arguments['<stage>'] == "production" else "f0077c516d",
+            "framework_slug": "digital-outcomes-and-specialists-2"
+        },
+        {
+            "lot_slug": "user-research-participants",
+            "list_id": "prod_list_id3" if arguments['<stage>'] == "production" else "d35601203b",
```

Comment:
Great point, it is a bit nasty (but on the plus side fairly contained in it's scope to the `lots` dict).

We could remove the if statement based on environment by doing something like:
```
CONFIG = {
    "preview": {
        "digital-specialists": "list id",
        "digital-outcomes": "list id2",
        "user-research-participants": "list id3",
    },
    "production": {
        "digital-specialists": "list id4",
        "digital-outcomes": "list id5",
        "user-research-participants": "list id6",
    }
} # would have all stages in real code
```

```
lots = [
        {
            "lot_slug": "digital-specialists",
            "list_id": CONFIG[arguments['<stage>']]['digital-specialists'],
            "framework_slug": "digital-outcomes-and-specialists-2"
        }
] # would have all lots in real code
```

It would be a bit nicer but without moving list ID's into the database or storing them in credentials repo for every stage for every lot and passing them into the script via jenkins instead we still have a mix of code and policy.

Do you feel the current solution is tolerable or it's worth looking into any of the above?